azafea/event_processors/endless/metrics/v3/model/__init__.py

```diff
@@ -570,6 +570,64 @@ def _get_fields_from_payload(payload: GLib.Variant) -> Dict[str, Any]:
         }
 
 
+class SplitFlatpakRepoStats(SingularEvent):
+    """Statistics for splitting the Flatpak repo from the OSTree repo.
+
+    Prior to Endless OS 4, both the OS and Flatpaks were stored in a
+    single, shared OSTree repository. On the first boot into Endless OS
+    4 or later, these are split into two repositories, one for the OS
+    and one for Flatpaks. This can take a long time, and blocks the boot
+    process, so we report this event to quantify how long it takes and
+    why.
+
+    This is a one-time operation which only runs if needed, so this
+    event is sent at most once per Endless OS system.
+
+    :UUID name: ``SPLIT_FLATPAK_REPO_STATS`` in eos-boot-helper
+
+    .. versionadded:: 4.0.1
+    """
+    __tablename__ = 'split_flatpak_repo_stats_v3'
+    __event_uuid__ = '880fd091-2e68-4bd2-baa1-f6810fabcc1c'
+    __payload_type__ = '(uuuuuuutut)'
+
+    #: time elapsed in seconds
+    elapsed = Column(Integer, nullable=False)
+    #: number of OS OSTree refs duplicated
+    num_os_refs = Column(Integer, nullable=False)
+    #: number of Flatpak OSTree refs duplicated
+    num_flatpak_refs = Column(Integer, nullable=False)
+    #: number of other OSTree refs duplicated
+    num_other_refs = Column(Integer, nullable=False)
+    #: number of OSTree objects duplicated
+    num_objects = Column(Integer, nullable=False)
+    #: number of OSTree static deltas duplicated
+    num_deltas = Column(Integer, nullable=False)
+    #: number of Flatpak apps duplicated
+    num_apps = Column(Integer, nullable=False)
+    #: size of Flatpak apps duplicated in bytes
+    size_apps = Column(BigInteger, nullable=False)
+    #: number of Flatpak runtimes duplicated
+    num_runtimes = Column(Integer, nullable=False)
+    #: size of Flatpak runtimes duplicated in bytes
+    size_runtimes = Column(BigInteger, nullable=False)
+
+    @staticmethod
+    def _get_fields_from_payload(payload: GLib.Variant) -> Dict[str, Any]:
+        return {
+            'elapsed': payload.get_child_value(0).get_uint32(),
```

Comment:
If you were to change the type of this field to `Interval` you would want this change I think:

```suggestion
            'elapsed': datetime.timedelta(seconds=payload.get_child_value(0).get_uint32()),
```